src/main/java/org/inferred/freebuilder/processor/Metadata.java

```diff
@@ -135,6 +135,12 @@ public ParameterizedType getBuilder() {
   /** Returns metadata about the properties of the type. */
   public abstract ImmutableList<Property> getProperties();
 
+  /** Returns super types implementing builder */
+  public abstract ImmutableSet<ParameterizedType> getSuperBuilderTypes();
+
+  /** Returns super type properties */
+  public abstract ImmutableMap<ParameterizedType, ImmutableList<Property>> getSuperTypeProperties();
```

Comment:
I'd prefer this map stored a new value type containing a set of properties and a boolean indicating whether there is a builder or not.